### PR TITLE
nix-serve: unstable-2018-03-20 → unstable-2024-09-17

### PR DESCRIPTION
**Pull request by fabianhjr**
## Description of changes

Draft while upstream decides way forward

https://github.com/edolstra/nix-serve/pull/60

Fixes broken nixosTests.nix-serve after

https://github.com/NixOS/nixpkgs/pull/335342

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabianhjr** [comment]
@ofborg test nix-serve

**Comment by Aleksanaa** [comment]
We should probably pin to 2.18 if it isn't merged in one or two days. Better not send a broken nix-serve to nixos-unstable

**Comment by fabianhjr** [comment]
pinning nix to 2.18 for this package also sounds like an acceptable solution, we could even create a PR and merge that meanwhile + rebase this to remove that pinning.

**Comment by fabianhjr** [comment]
@ofborg test nix-serve

**Comment by Mic92** [comment]
alternative approach: https://github.com/NixOS/nixpkgs/pull/342754

**Comment by fabianhjr** [comment]
Upstream merged, leaving as draft due to alternative offered by @Mic92 though not closing for the time being.

File: pkgs/tools/package-management/nix-serve/default.nix [hunk 1]
```diff
@@ -9,8 +9,8 @@
 }:
 
 let
-  rev = "e4675e38ab54942e351c7686e40fabec822120b9";
-  sha256 = "1wm24p6pkxl1d7hrvf4ph6mwzawvqi22c60z9xzndn5xfyr4v0yr";
+  rev = "06c1e229ac647cf9733af509e83a8407c6141006";
```
Comment (Mic92) [inline pkgs/tools/package-management/nix-serve/default.nix:12]:
```suggestion
  rev = "77ffa33d83d2c7c6551c5e420e938e92d72fec24";
```

**Comment by fabianhjr** [comment]
This will be berojen due to the nix 2.24 revert :s

**Comment by Mic92** [comment]
Follow up: https://github.com/NixOS/nixpkgs/pull/343274

**Pull request merged**